### PR TITLE
Mod: don't use read queue

### PR DESCRIPTION
**Pull request by fangyincheng**

### CODE REVIEW

File: session.go [hunk 1]
```diff
@@ -65,8 +65,7 @@ type session struct {
 	reader Reader // @reader should be nil when @conn is a gettyWSConn object.
 	writer Writer
 
-	// read & write
-	rQ chan interface{}
+	// write
```
Comment (AlexStocks) [inline session.go]: Hey, guy, u can add "// DEPRECATED" on this func, pls do not delete it. U can refer to https://github.com/alexstocks/getty/blob/master/getty.go#L153.

File: session.go [hunk 2]
```diff
@@ -274,18 +273,6 @@ func (s *session) SetCronPeriod(period int) {
 	s.period = time.Duration(period) * time.Millisecond
 }
 
-// set @session's read queue size
-func (s *session) SetRQLen(readQLen int) {
```
Comment (AlexStocks) [inline session.go]: pls do not delete it. just leave a empty func body is enough to compatible with previous version.

File: session.go (hunk 1 above)
Comment (fangyincheng) [reply session.go]: done

File: session.go (hunk 2 above)
Comment (fangyincheng) [reply session.go]: done

**Comment by wongoo** [comment]
@fangyincheng conflict need to resolve

File: session.go [hunk 3]
```diff
@@ -522,7 +507,7 @@ LOOP:
 		case <-s.done:
 			// this case branch assure the (session)handleLoop gr will exit before (session)handlePackage gr.
 			if atomic.LoadInt32(&(s.grNum)) == 1 { // make sure @(session)handlePackage goroutine has been closed.
-				if len(s.rQ) == 0 && len(s.wQ) == 0 {
+				if len(s.wQ) == 0 {
 					log.Infof("%s, [session.handleLoop] got done signal. Both rQ and wQ are nil.", s.Stat())
```
Comment (wongoo) [inline session.go]: log content must be updated

File: session.go [hunk 4]
```diff
@@ -573,14 +546,18 @@ LOOP:
 }
 
 func (s *session) addTask(pkg interface{}) {
+	f := func() {
+		s.listener.OnMessage(s, pkg)
+		s.incReadPkgNum()
+	}
 	if s.tPool != nil {
 		s.tPool.AddTask(func() {
```
Comment (wongoo) [inline session.go]: replace the parameter of `AddTask` with `f`

Comment (AlexStocks) [reply session.go]: @wongoo u always do a  great review job.

Comment (fangyincheng) [reply session.go]: sorry，it was my fault.

File: session.go (hunk 3 above)
Comment (fangyincheng) [reply session.go]: done

File: session.go [hunk 5]
```diff
@@ -522,8 +507,8 @@ LOOP:
 		case <-s.done:
 			// this case branch assure the (session)handleLoop gr will exit before (session)handlePackage gr.
 			if atomic.LoadInt32(&(s.grNum)) == 1 { // make sure @(session)handlePackage goroutine has been closed.
-				if len(s.rQ) == 0 && len(s.wQ) == 0 {
-					log.Infof("%s, [session.handleLoop] got done signal. Both rQ and wQ are nil.", s.Stat())
+				if len(s.wQ) == 0 {
+					log.Infof("%s, [session.handleLoop] got done signal. Both wQ are nil.", s.Stat())
```
Comment (wongoo) [inline session.go]: wQ is nil.

Comment (fangyincheng) [reply session.go]: ok

**Comment by wongoo** [comment]
LGTM

**Pull request merged**